fog/report/validation/src/ingest_report.rs

```diff
@@ -24,23 +21,10 @@ impl IngestReportVerifier {
         &self,
         remote_report: VerificationReport,
     ) -> Result<RistrettoPublic, Error> {
-        let verification_report_data = VerificationReportData::try_from(&remote_report)?;
-        let report_data: ReportData = verification_report_data.quote.report_body()?.report_data();
+        let parsed_report = self.verifier.verify(&remote_report)?;
+        let report_data = parsed_report.quote.report_body()?.report_data();
         let report_data_bytes: &[u8] = report_data.as_ref();
-
-        // Extract the pubkey from the signed evidence
-        let report_pubkey: RistrettoPublic = RistrettoPublic::try_from(&report_data_bytes[32..64])?;
-
-        let masked_report_data = ReportDataMask::new_with_mask(report_data_bytes, &[0u8; 32])?;
-
-        let mut verifier = self.verifier.clone();
-        verifier
-            .sign_type(QuoteSignType::Linkable)
-            .report_data(&masked_report_data);
-
-        verifier.verify(&remote_report)?;
-
-        Ok(report_pubkey)
+        Ok(RistrettoPublic::try_from(&report_data_bytes[32..64])?)
```

Comment:
Is there a guarantee there are 64 bytes in the report data?

Reply:
Yeah, it's a fixed length in the SGX SDK, and they do a bunch of "cast these bytes to this structure"-style C, so if they did change this it would be a severe breaking change of the SGX SDK.